test/e2e/set/scyllacluster/verify.go

```diff
@@ -83,9 +83,124 @@ func verifyPodDisruptionBudget(sc *scyllav1.ScyllaCluster, pdb *policyv1.PodDisr
 func verifyScyllaCluster(ctx context.Context, kubeClient kubernetes.Interface, sc *scyllav1.ScyllaCluster) {
 	framework.By("Verifying the ScyllaCluster")
 
+	sc = sc.DeepCopy()
+
+	o.Expect(sc.CreationTimestamp).NotTo(o.BeNil())
 	o.Expect(sc.Status.ObservedGeneration).NotTo(o.BeNil())
+	o.Expect(*sc.Status.ObservedGeneration).To(o.BeNumerically(">=", sc.Generation))
 	o.Expect(sc.Status.Racks).To(o.HaveLen(len(sc.Spec.Datacenter.Racks)))
 
+	for i := range sc.Status.Conditions {
+		c := &sc.Status.Conditions[i]
+		o.Expect(c.LastTransitionTime).NotTo(o.BeNil())
+		o.Expect(c.LastTransitionTime.Time.Before(sc.CreationTimestamp.Time)).NotTo(o.BeTrue())
+
+		// To be able to compare the statuses we need to remove the random timestamp.
+		c.LastTransitionTime = metav1.Time{}
+	}
+	o.Expect(sc.Status.Conditions).To(o.ConsistOf(func() []interface{} {
+		type condValue struct {
+			condType string
+			status   bool
+		}
+		condList := []condValue{
+			// Aggregated conditions
+			{
+				condType: "Available",
+				status:   true,
+			},
+			{
+				condType: "Progressing",
+				status:   false,
+			},
+			{
+				condType: "Degraded",
+				status:   false,
+			},
+
+			// Controller conditions
+			{
+				condType: "ServiceAccountControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "ServiceAccountControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "RoleBindingControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "RoleBindingControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "AgentTokenControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "AgentTokenControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "StatefulSetControllerAvailable",
+				status:   true,
+			},
+			{
+				condType: "StatefulSetControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "StatefulSetControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "ServiceControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "ServiceControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "PDBControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "PDBControllerDegraded",
+				status:   false,
+			},
+			{
+				condType: "IngressControllerProgressing",
+				status:   false,
+			},
+			{
+				condType: "IngressControllerDegraded",
+				status:   false,
```

Comment:
wouldn't it be better to use metav1.ConditionTrue/False here without need for convertion in loop below?

Reply:
done